### PR TITLE
Add back validation on process/next

### LINKED ISSUE #470: Validation does not run on process/next in v8
### Description of the bug

Reported on slack

### Steps To Reproduce

> Hei :slightly_smiling_face:
> Jeg tester ut v8 og får ikke valideringen til å fungere ved process/next, mellom task_1 og task_2 som jeg er vant til fra v7.
Eks. Jeg har lagt til en dataType i applicationMetadata og satt MinCount: 1 og taskId: "Task_1". Når jeg går videre til task_2 med process/next så hadde jeg forventet at den  ikke skulle gå videre til task_2, dersom datatypen med MinCount: 1 inne var lastet opp i Task_1
> Er dette en feil, eller kommer det noe nytt i v8 rundt validering av task ved process next?
> Det ser ut som dere gjør validering i completeProcess, men ikke mollom hver task.

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ivarne**
Also use ProblemDetails for returning errors in process controller and return the validationIssues for frontend when relevant.

## Related Issue(s)
- https://github.com/Altinn/app-lib-dotnet/issues/470

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.App.Api/Controllers/ProcessController.cs [hunk 1]
```diff
@@ -227,42 +245,55 @@ private async Task<bool> CanTaskBeEnded(Instance instance, string currentTaskId,
...
-                ProcessNext? processNext = null;
-                if (Request.Body != null && Request.Body.CanRead)
-                {
-                    processNext = await DeserializeFromStream<ProcessNext>(Request.Body);
-                }
-
                 Instance instance = await _instanceClient.GetInstance(app, org, instanceOwnerPartyId, instanceGuid);
 
-                if (instance?.Process == null)
+                var currentTaskId = instance.Process.CurrentTask?.ElementId;
```
Comment (bjorntore) [inline src/Altinn.App.Api/Controllers/ProcessController.cs:255]: Super nitpicky, can ignore: Use explicit datatype when you can't tell the type 100% from the right side.

**Review by bjorntore — APPROVED** [review]

**Review by RonnyB71 — APPROVED** [review]

File: src/Altinn.App.Api/Controllers/ProcessController.cs (hunk 1 above)
Comment (ivarne) [reply src/Altinn.App.Api/Controllers/ProcessController.cs:255]: There seems to be a style-rule that matches that expectation. 

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0007-ide0008#csharp_style_var_when_type_is_apparent

File: src/Altinn.App.Api/Controllers/ProcessController.cs [hunk 2]
```diff
@@ -299,6 +330,41 @@ private async Task<bool> CanTaskBeEnded(Instance instance, string currentTaskId,
             }
         }
 
+        private ActionResult<AppProcessState> GetResultForError(ProcessChangeResult result)
```
Comment (tjololo) [inline src/Altinn.App.Api/Controllers/ProcessController.cs:333]: Wouldn't it be more correct with:
```suggestion
        private ActionResult<ProblemDetails> GetResultForError(ProcessChangeResult result)
```

**Review by tjololo — APPROVED** [review]
Looks good. Just a nitpicks

File: src/Altinn.App.Api/Controllers/ProcessController.cs (hunk 2 above)
Comment (ivarne) [reply src/Altinn.App.Api/Controllers/ProcessController.cs:333]: Dette funka ikke, reverter.

**Pull request merged**